lib/rules/no-misleading-character-class.js

```diff
@@ -182,11 +200,22 @@ module.exports = {
                     }];
                 }
 
-                context.report({
-                    node,
-                    messageId: kind,
-                    suggest
-                });
+                for (const char of chars) {
+                    context.report({
+                        loc: {
+                            end: {
+                                line: node.loc.end.line,
+                                column: node.loc.start.column + char.end
+                            },
+                            start: {
+                                line: node.loc.start.line,
+                                column: node.loc.start.column + char.start
+                            }
+                        },
+                        messageId: kind,
+                        suggest
+                    });
+                }
```

Comment:
In theory, we could add logic to "merge" neighboring `char` reports of the same `kind`. For example, the `/[👶🏻]/` from the PR description could give a single report on col 12-14 instead of one on 12-13 and one on col 13-14.

I held off to avoid extra complexity & so the rule gives more granular info, but it is tempting... thoughts?

Reply:
👍 - fixed the columns, and will keep the two reports.